haystack/pipelines/base.py

```diff
@@ -853,6 +857,11 @@ def get_document_store(self) -> Optional[BaseDocumentStore]:
         :return: Instance of DocumentStore or None
         """
         matches = self.get_nodes_by_class(class_type=BaseDocumentStore)
+        if len(matches) == 0:
+            matches = list(
+                set([retriever.document_store for retriever in self.get_nodes_by_class(class_type=BaseRetriever)])
+            )
+
         if len(matches) > 1:
             raise Exception(f"Multiple Document Stores found in Pipeline: {matches}")
```

Comment:
I know this is not part of your changes, but why having more than a docstore is a problem? I can imagine forked pipelines with two retrievers and a JoinDocuments node before the Reader could make sense.

In addition: if the exception should stay, then it's a `PipelineError` :smile:

Reply:
I think the exception still makes sense in case there are really two different docstores in the game. In such a case the user has to dig out the needed docstore herself as we cannot give any general infos to identify them easily. Most of the time this should not be an issue in multi-retriever settings anyway: the case that you depicted should use only one instance of the docstore. That should be safely handled by the implicit unique operation executed by set(). But I'm gonna write a test for that case to ensure :-)

Regarding the PipelineError, yes I can understand it, but let's introduce this in a separate PR. I think this will be a bigger one to get it clean.